lib/OpenQA/Worker/Jobs.pm

```diff
@@ -339,7 +339,7 @@ sub upload_status(;$) {
         }
         $current_running = $os_status->{running};
     }
-    if ($upload_result) {
+    if (defined($upload_result)) {
         $status->{result} = read_result_file($upload_result);
         if ($os_status->{running}) {
             $status->{result}->{$os_status->{running}} //= {};
```

Comment:
So then $status->{result}->{''} is allowed? How that will be interpreted by scheduler?